### ISSUE TITLE
Add more infos to community map popup

### ISSUE THREAD

**Issue by zyzo** [fossasia/api.fossasia.net#20]
Integrate calendar service with the community map by adding community latest/upcoming event on the map marker

**Comment by mariobehling** [comment]
Integrating a calendar, we could also think of integrating complete events in the re-data format (https://github.com/opendatacity/re-data) and services like open-event (https://github.com/fossasia/open-event).

**Comment by andibraeu** [comment]
you can also display the latest blog posts from their rss feeds

**Comment by zyzo** [comment]
To display events, it's not possible to query the CalendarAPI at loading phase, because there are a lot of communities and it will be 1 api call per community. There are, as I know, two solutions to address this : 

* store the events in `ffGeojson.json`
* embed the [timeline](https://github.com/fossasia/timeline.api.fossasia.net) as an `iframe` inside popup. That way it will only load when opened.

I vote for  solution 2 for code-reuse and dynamicity. But that also means we have to deploy an embed timeline service.

**Comment by zyzo** [comment]
Added using embed timeline http://cmap-fossasia-api.herokuapp.com/

**Comment by mariobehling** [comment]
Please provide a screenshot of the embedded timeline. I am not clear where can I find it. Which one is it?

**Comment by zyzo** [comment]
![screenshot from 2015-06-23 13 08 15](https://cloud.githubusercontent.com/assets/6483446/8426136/6c2dc84e-1f0d-11e5-8b38-ed6843131649.png)

**Comment by mariobehling** [comment]
I don't understand what is it "July 17 and June 19"? It is from another community. How is this related to Novosibirsk?

**Comment by zyzo** [comment]
This is because not a lot of fossasia communities have ics feed yet, so the data used is from freifunk for demo purpose. This feature is currently more useful at freifunk given the dataset.

**Comment by mariobehling** [comment]
Ok, thanks for the demo. Is it working in the freifunk map then? The show up of July 17 and July 19. I am confused by this demo as the same data shows up for all communities. A demo that would make more sense to me is, if every community had different data. Now, I do not know if it is always like that. Do we always have the same data for all communities or not? Another demo would make more sense to be clear of the outcome of this feature implementation.

If you say it works for freifunk, why don't you provide a corresponding freifunk screenshot?

**Comment by zyzo** [comment]
No I'm not saying that it's working at freifunk. 

Normally importing developed features to freifunk is handled by @andibraeu, but surely I can clone the [freifunk repo](https://github.com/freifunk/cmap.api.freifunk.net) to integrate this feature and show a demo that might make more sense. Do you want me to do that ?

**Comment by mariobehling** [comment]
yes, please. We want to see the cool stuff that comes out of it and test it in a real world setting.

**Comment by zyzo** [comment]
Freifunk demo : http://etud.insa-toulouse.fr/~hadang/freifunk-api/cmap.api.freifunk.net/map.html

Some communities with data to test : 
Events :

* Freifunk Wuppertal
* Freifunk 3Landereck
* weimarnetz

Blog posts :

* Freifunk Augsburg
* Freifunk Göttingen
* Freifunk Chemnitz e.V.

**Comment by andibraeu** [comment]
Am Mittwoch, den 01.07.2015, 01:21 -0700 schrieb DANG HAI AN:
> Freifunk demo : http://etud.insa-toulouse.fr/~hadang/freifunk
> -api/cmap.api.freifunk.net/map.html
> Some communities with data to test : 
> Events :
> Freifunk Wuppertal
> Freifunk 3Landereck
> weimarnetz
> Blog posts :
> Freifunk Augsburg
> Freifunk Göttingen
> Freifunk Chemnitz e.V.
> —
> 

That looks really good. Unfortunately it doesn't work on firefox for
me. Can you test that please? For Weimarnetz an event from yesterday
shows up.

Two design issues:
* could you make a bulletin list of the recent blogposts?
* Could you reduce the width of the popup boxes if there's no event?
Did you try to set the events one below the other?

**Comment by zyzo** [comment]
>  Unfortunately it doesn't work on firefox for
me. 

Could you specify what didn't work and how to reproduce it ?

>  For Weimarnetz an event from yesterday
shows up.

This is the expected behaviour, what I want is to display the latest event and the closest upcoming event. We could discuss whether to diplay just the upcoming events as well

> Two design issues:
> * could you make a bulletin list of the recent blogposts?
> * Could you reduce the width of the popup boxes if there's no event?
> Did you try to set the events one below the other?

I'll work on this

**Comment by zyzo** [comment]
The demo is update with previous updates
